src/screen.js

```diff
@@ -11,9 +11,15 @@ const AlpineScreenMagicMethod = {
                 update = setTimeout(() => {
                     $el.__x.updateElements($el)
                 }, 150)
+
+                // set $screenHelperInitialized to prevent multiple calls
+                window.$screenHelperInitialized = true
```

Comment:
does it need to be on the window? can't we just have a variable on the closure?

Reply:
We use `$el.__x.updateElements($el)` where it re-runs $screen helper each time windows resized. Which means everything starts from 0. I am not sure if this is expected behavior tho. All i am trying to do is observe screen width and update the breakpoints.

Reply:
I mean that this function/module (screen.js) has its own scope, we could store the listener/whether it's been initialised in the module instead of putting it on the window

Reply:
hmm, you mean something like this?

```js
let Initialized = false;

const AlpineScreenMagicMethod = {
    start() {
        Alpine.addMagicProperty('screen', function ($el) {
            // bind resize event to window with debounce
            let update
            const updateScreen = () => {
                clearTimeout(update)
                update = setTimeout(() => {
                    $el.__x.updateElements($el)
                }, 150)

                Initialized = true
            }
            // bind resize event to window if not initialized
            if (!Initialized) {
                window.addEventListener('resize', updateScreen)
            }

....

```

Reply:
i am trying to achieve this but still memory leaking.

```js
            let update

            window.addEventListener('resize', updateScreen => {

              clearTimeout(update)

              update = setTimeout(() => {
                  $el.__x.updateElements($el)
              }, 150)

              window.removeEventListener('resize', updateScreen, false)

            }, false)
```

Reply:
hmm what we're likely going to need to do is keep track of which components need to be updated.

something like:
```js
const screenComponents = [];
function debouncedScreensUpdate() {
    let update
    const updateScreens = () => {
        clearTimeout(update)
        update = setTimeout(() => {
            screenComponents.forEach(($el) => 
                $el.__x && $el.__x.updateElements($el)
            )
         }, 150)
     };
    return updateScreens;
}
let initialize = false;
const AlpineScreenMagicMethod = {
    start() {
        Alpine.addMagicProperty('screen', function ($el) {
            screenComponents.push($el);
            if (!initialized) {
                window.addEventListener('resize', debouncedScreensUpdate())
                initialized = true;
            }
```

Reply:
with something like the above, we can attach 1 resize handler but update all the components that use `$screen`.

Unless I misunderstand how `addMagicProperty` property callback works

Reply:
we could even move the `addEventListener` stuff to the `start()` function (outside of the `addMagicProperty` callback) so we don't even need to keep track of whether the listener is initialised or not

(`start()` should be called once vs the `addMagicProperty` callback which is called for every instance)

Reply:
That should work. Screencomponents could be a Set() rather than an array so we won't get duplications if you use the helper twice in the same component

Reply:
https://codepen.io/hugodf/pen/mdEZqaB?editors=1111 made this to clear up for me: `addMagicProperty` callback gets called on every `$screen` reference/invocation.

Here's a pared down version of "screen" working with the approach I mentioned above (1 resize listener, an array to keep track of elements to update) --> https://codepen.io/hugodf/pen/oNLroVB

Reply:
So probably what we can do in the end is:
```js
const screenComponents = [];
function debouncedScreensUpdate() {
    let update
    const updateScreens = () => {
        clearTimeout(update)
        update = setTimeout(() => {
            screenComponents.forEach(($el) => 
                $el.__x && $el.__x.updateElements($el)
            )
         }, 150)
     };
    return updateScreens;
}
let initialize = false;
const AlpineScreenMagicMethod = {
    start() {
        window.addEventListener('resize', debouncedScreensUpdate())
        Alpine.addMagicProperty('screen', function ($el) {
            screenComponents.push($el);
```

Reply:
We can do something like: Inside the foreach, if document doesn't contain $el, we remove $el from the list otherwise we trigger an update. This is also necessary for stuff like turbolinks.